File: angular/aisentona-web/src/app/services/weather.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { WeatherResponse } from '../core/interfaces/Response/Weather';

@Injectable({
  providedIn: 'root'
})
export class WeatherService {
  private apiUrl = 'https://api.met.no/weatherapi/locationforecast/2.0/compact';

  constructor(private http: HttpClient) { }

  getWeather(latitude: number, longitude: number): Observable<WeatherResponse> { {
    const headers = { 'User-Agent': 'VeritareNews/1.0 (rodrigofreires05@gmail.com)' };

    const url = `${this.apiUrl}?lat=${latitude}&lon=${longitude}`;

    return this.http.get<WeatherResponse>(url, { headers });
  }
}

}

File: Aisentona.Biz/Services/WeatherService.cs
using Aisentona.Entities.Response;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace Aisentona.Biz.Services
{
    public class WeatherService
    {
        private readonly HttpClient _httpClient;

        public WeatherService()
        {
            _httpClient = new HttpClient();
            _httpClient.DefaultRequestHeaders.Add("User-Agent", "SeuNomeOuEmail"); // Obrigatório para Met.no
        }

        public async Task<WeatherResponse> GetWeatherAsync(double latitude, double longitude)
        {
            string url = $"https://api.met.no/weatherapi/locationforecast/2.0/compact?lat={latitude}&lon={longitude}";

            HttpResponseMessage response = await _httpClient.GetAsync(url);

            if (!response.IsSuccessStatusCode)
            {
                throw new Exception($"Erro ao obter previsão do tempo: {response.StatusCode}");
            }

            string jsonResponse = await response.Content.ReadAsStringAsync();
            return JsonSerializer.Deserialize<WeatherResponse>(jsonResponse, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
        }
    }
}
